fabtests/common/shared.c

```diff
@@ -578,18 +597,14 @@ int ft_alloc_msgs(void)
 		if (ret)
 			return ret;
 	}
+
 	ret = ft_hmem_memset(opts.iface, opts.device, (void *) buf, 0, buf_size);
 	if (ret)
 		return ret;
-	rx_buf = buf;
-
-	if (opts.options & FT_OPT_ALLOC_MULT_MR)
-		tx_buf = (char *) buf + MAX(rx_size, FT_MAX_CTRL_MSG);
-	else
-		tx_buf = (char *) buf +
-			 MAX(rx_size, FT_MAX_CTRL_MSG) * opts.window_size +
-			 rma_resv_bytes;
 
+	rx_buf = (char *)ft_get_aligned_addr(buf, alignment);
+	assert((rx_buf - buf) >= 0 && (rx_buf - buf) < alignment);
+	tx_buf = buf + half_buf_size;
```

Comment:
Similar comment as before.  Why is rx_buf aligned, but tx_buf is not?

Reply:
because `half_buf_size` is returned from `get_aligned_size`. So, if `rx_buf` is aligned, `tx_buf` must be aligned.

Reply:
I think you want
```tx_buf = rx_buf + half_buf_size```
not *buf* + half_size.  buf isn't guaranteed to be aligned.

Reply:
Yes. you are right. Fixed in new revision.